test/erc-20-extensions/ERC20Extensions.js

```diff
@@ -252,3 +254,49 @@ describe('ERC20ExtensionsMock tests', function () {
     })
   })
 })
+
+// Transaction needs to be propagated to the mirror node
+async function pollForChangedSupply(ERC20Burnable, initialSupply){
+  let numberOfTries = 0;
+  const timesToTry = 300;
+  let newSupply = 0;
+
+  while (numberOfTries < timesToTry) {
+
+    newSupply = await ERC20Burnable.totalSupply()
```

Comment:
Q: what if the supply changes before this line is executed, in that case you'll never see a change.
Maybe consider passing in the original balance, this will ensure tests get the initialSupply prior to whatever execution they kick off

Reply:
Would that not assume the initialSupply and initialBalance are the same?  Balance is bound to an address.

Reply:
Not at all.
A test would be responsible for getting the balance for the said address. It would then pass the value in to verify that it had not changed.
If not this or a similar approach i'm not sure how you ensure you never fail in the race condition in which an update may occur prior to the test logic getting run. Is this possible?